src/main/java/uk/gov/ons/census/fwmt/csvservice/implementation/addresscheck/AddressCheckConverterService.java

```diff
@@ -77,19 +89,48 @@ public void convertToCanonical() throws GatewayException {
-      if (postcodeLookupMap.containsKey(addressCheckListing.getPostcode().replaceAll("\\s+","").toUpperCase())) {
-        CreateFieldWorkerJobRequest createFieldWorkerJobRequest = createAddressCheckJob(addressCheckListing,
-            postcodeLookupMap.get(addressCheckListing.getPostcode().replaceAll("\\s+","").toUpperCase()));
-        gatewayActionAdapter.sendJobRequest(createFieldWorkerJobRequest, CANONICAL_ADDRESS_CHECK_CREATE_SENT);
-        gatewayEventManager.triggerEvent(String.valueOf(createFieldWorkerJobRequest.getCaseId()),
-            CSV_ADDRESS_CHECK_REQUEST_EXTRACTED);
-      } else {
+      if (postcodeLookupMap.containsKey(getPostcode(addressCheckListing))) {
+        if (addressCheckListingIsValid(postcodeLookupMap.get(getPostcode(addressCheckListing)))) {
+          CreateFieldWorkerJobRequest createFieldWorkerJobRequest = createAddressCheckJob(addressCheckListing,
+              postcodeLookupMap.get(getPostcode(addressCheckListing)));
+          gatewayActionAdapter.sendJobRequest(createFieldWorkerJobRequest, CANONICAL_ADDRESS_CHECK_CREATE_SENT);
+          gatewayEventManager.triggerEvent(String.valueOf(createFieldWorkerJobRequest.getCaseId()),
+              CSV_ADDRESS_CHECK_REQUEST_EXTRACTED);
+        } else {
+          RejectionReport rejectionReport = getRejectionReport(addressCheckListing.getCaseReference(),
+              "Lookup file missing data");
+          rejectedReportAddressCheckListing.add(rejectionReport);
+          rejectedAddressCheckListing.add(addressCheckListing);
+          gatewayEventManager
+              .triggerErrorEvent(this.getClass(), "Postcode: " + addressCheckListing.getPostcode(),
+                  "N/A", LOOKUP_FILE_MISSING_DATA);
+        }
+      } else if (!postcodeLookupMap.containsKey(getPostcode(addressCheckListing))) {
+        RejectionReport rejectionReport = getRejectionReport(addressCheckListing.getCaseReference(),
+            "Failed to match postcode: " + addressCheckListing.getPostcode());
+        rejectedReportAddressCheckListing.add(rejectionReport);
+        rejectedAddressCheckListing.add(addressCheckListing);
         gatewayEventManager
-            .triggerErrorEvent(this.getClass(),"Postcode: " + addressCheckListing.getPostcode(),
-                "N/A", GatewayEventsConfig.FAILED_MATCH_POSTCODE);
+            .triggerErrorEvent(this.getClass(), "Postcode: " + addressCheckListing.getPostcode(),
+                "N/A", FAILED_MATCH_POSTCODE);
       }
     }
   }
+
+  private String getPostcode(AddressCheckListing addressCheckListing) {
+    return addressCheckListing.getPostcode().replaceAll("\\s+", "").toUpperCase();
+  }
+
+  private boolean addressCheckListingIsValid(PostcodeLookup postcodeLookup) {
+    return !postcodeLookup.getAreaRoleId().isBlank() && !postcodeLookup.getLa().isBlank() && !postcodeLookup.getLaName()
+        .isBlank();
+  }
+
+  private RejectionReport getRejectionReport(String caseRef, String reason) {
```

Comment:
Actually this isnt a get its a create, so rename to 
getRejectionReportEntry.

Also its not a RejectionReport but a RejectionReportEntry (so rename)